packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -299,6 +254,55 @@ function handleComplexSelector(element: Element, selector: Complex): Element[] {
   return elements;
 }
 
+/**
+ * Handles compound selectors or a list of selectors for subjective element.
+ * To handle multi-paths matching, this function create branches internally and validates each cases.
+ * @param element The subjective element
+ * @param selectors A list of selectors to sequentially validated
+ * @returns A list of elements matches the list of selectors
+ */
+function handleCompoundSelector(element: Element, selectors: AST[]): Element[] {
+  if (selectors.length === 0) {
+    return [];
+  }
+
+  // Start with validating the subjective element with given selectors
+  const branches: [Element, number][] = querySelectorAll(element, selectors[0]).map(
```

Comment:
I think, the assumption that the first selector is not a transposition is sound. So, only as a comment (so I'm not pushing to change it).

If we would need to support selector like "div :upwind(1)" (perhaps there are use cases for other types of transpositions, because for "upwind" I cannot think of real-world use-cases), it would be still possible without changing much. (Basically, adding a special case that detects it, and starting only with a single brach with `[element, 0]` should work if I'm not mistaken).